tests/test_mig_shared_functionality_cat.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+#
+# --- BEGIN_HEADER ---
+#
+# test_mig_shared_functionality_cat - cat functionality unit test
+# Copyright (C) 2003-2024  The MiG Project by the Science HPC Center at UCPH
+#
+# This file is part of MiG.
+#
+# MiG is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# MiG is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+#
+# --- END_HEADER ---
+#
+
+"""Unit tests of the MiG functionality file implementing the cat resource."""
+
+from __future__ import print_function
+import importlib
+import os
+import shutil
+import sys
+
+from tests.support import MIG_BASE, MigTestCase, testmain, \
+    fixturefile, fixturefile_normname, \
+    _ensuredirs, _temppath
+
+from mig.shared.base import client_id_dir
+from mig.shared.functionality.cat import _main as main
+
+
+def create_http_environ(configuration, wsgi_variables={}):
+    """Small helper that can create a minimum viable environ dict suitable
+    for passing to http-facing code for the supplied configuration."""
+
+    environ = {}
+    environ['MIG_CONF'] = configuration.config_file
+    environ['HTTP_HOST'] = wsgi_variables.get('http_host', 'localhost')
+    environ['PATH_INFO'] = wsgi_variables.get('path_info', '/')
+    environ['REMOTE_ADDR'] = wsgi_variables.get('remote_addr', '127.0.0.1')
+    environ['SCRIPT_URI'] = ''.join(('http://', environ['HTTP_HOST'], environ['PATH_INFO']))
+    return environ
+
+
+class MigCgibinCat(MigTestCase):
```

Comment:
Minor: another CGI leftover to strip.
I'll rename to fit unit test module name during merge and add the missing docstring while at it.